hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -882,9 +884,26 @@ bool MDMParser::_powerOn(void)
     bool retried_after_reset = false;
     int i = MDM_POWER_ON_MAX_ATTEMPTS_BEFORE_RESET; // When modem not responsive on boot, AT/OK tries 25x (for ~30s) before hard reset
 
-    // FIXME: REMOVE THIS BEFORE MERGING!!!!!!
-    _dev.dev = DEV_SARA_R510;
-    MDM_ERROR("Forcing to R510\r\n");
+    // TODO: 
+    // Implement `platform_ncp_electron` + `platform_primary_ncp_identifier` to determine NCP id? 
+    // -OR-
+    // Extend `deviceid_hal.h` ? 
+    uint8_t otp_ncp_id = 0;
+    int result = hal_get_device_ncp_id(&otp_ncp_id);
```

Comment:
Nit: might be a good idea to introduce a dedicated member method to get the modem type similar to `ncpId()` in cellular NCP classes to hide the logic of looking either at runtime `_dev.dev` or `platform_ncp_primary_indentifier()` to clean things up a bit.